python/lsst/daf/butler/_butler.py

```diff
@@ -1335,6 +1336,10 @@ def transfer_from(
         transfer_dimensions : `bool`, optional
             If `True`, dimension record data associated with the new datasets
             will be transferred.
+        dry_run : `bool`, optional
+            If `True` the transfer will be processed without any modifications
+            made to the target butler and as if the target butler did not
+            have any of the datasets.
```

Comment:
It will check to see whether the artifacts in the graph exist, right?

I agree with your comment on Jira that it'd be better if it didn't assume that the datasets were not in the target butler, but this is useful enough as a starting point.  Maybe move that qualification to a `Notes` section, and amend it to say that this might change in the future?

Reply:
It does everything except update target butler.